### PR TITLE
[ci] Add grouped tests & mark tests to run by default for fastcheck pipeline

### PR DESCRIPTION
**Pull request by khluu**
- Add grouped test for Async Engine, Inputs, Utils, Worker tests & Tensorizer, Metrics, Tracings tests
- Mark Regression, Basic Correctness, Core,  Distributed (4 GPU), and Entrypoints tests to run on fastcheck by default

Template for fastcheck pipeline is merged here: https://github.com/vllm-project/buildkite-ci/blob/main/scripts/test-template-fastcheck.j2

### CODE REVIEW

**Comment by DarkLight1337** [comment]
Is there a reason why we have to group the tests together rather than marking the individual tests `fast_check`?

**Comment by khluu** [comment]
> Is there a reason why we have to group the tests together rather than marking the individual tests `fast_check`?

It's mostly to save on launching instances & image pull overhead. Each job has to pull the Docker image which takes roughly 2-3 mins, while the test itself only runs for 3-4 mins... Grouping tests together like this can save like 10mins or more of billable time for each build

**Comment by DarkLight1337** [comment]
> It's mostly to save on launching instances & image pull overhead. Each job has to pull the Docker image which takes roughly 2-3 mins, while the test itself only runs for 3-4 mins... Grouping tests together like this can save like 10mins or more of billable time for each build

If that's the case, perhaps we should get rid of the individual tests in favour of grouping them together regardless of `fast_check`.

**Comment by khluu** [comment]
> > It's mostly to save on launching instances & image pull overhead. Each job has to pull the Docker image which takes roughly 2-3 mins, while the test itself only runs for 3-4 mins... Grouping tests together like this can save like 10mins or more of billable time for each build
> 
> If that's the case, perhaps we should get rid of the individual tests in favour of grouping them together regardless of `fast_check`.

Yup I was planning to do that even for full CI run too, but let's start with fastcheck first

File: .buildkite/test-pipeline.yaml [hunk 1]
```diff
@@ -7,8 +7,36 @@
...
+  - pytest -v -s metrics # Metrics
+  - "pip install \
+      opentelemetry-sdk \
+      opentelemetry-api \
+      opentelemetry-exporter-otlp \
+      opentelemetry-semantic-conventions-ai" # Tracings
+  - pytest -v -s tracing
+  - cd /vllm-workspace/test_docs/docs
+  - pip install -r requirements-docs.txt
+  - SPHINXOPTS=\"-W\" make html
```
Comment (DarkLight1337) [inline .buildkite/test-pipeline.yaml:35]:
```suggestion
- label: Tensorizer, Metrics, Tracing, Docs test
  fast_check: true
  fast_check_only: true
  commands:
  - apt-get install curl libsodium23 && pytest -v -s tensorizer_loader # Tensorizer
  - pytest -v -s metrics # Metrics
  - "pip install \
      opentelemetry-sdk \
      opentelemetry-api \
      opentelemetry-exporter-otlp \
      opentelemetry-semantic-conventions-ai" # Tracing
  - pytest -v -s tracing
  - cd /vllm-workspace/test_docs/docs # Docs
  - pip install -r requirements-docs.txt
  - SPHINXOPTS=\"-W\" make html
```

Comment (DarkLight1337) [reply .buildkite/test-pipeline.yaml:35]: Just a small nit.

Comment (DarkLight1337) [reply .buildkite/test-pipeline.yaml]: I guess that works too.

**Review by DarkLight1337 — APPROVED** [review]

File: .buildkite/test-pipeline.yaml (hunk 1 above)
Comment (khluu) [reply .buildkite/test-pipeline.yaml]: fixed. Thanks!

Comment (khluu) [reply .buildkite/test-pipeline.yaml]: Actually let's split doc build to run individually since it's not running on GPU machine like other tests

Comment (khluu) [reply .buildkite/test-pipeline.yaml]: ^ and it uses small CPU instance so cost shouldn't matter

File: .buildkite/test-pipeline.yaml [hunk 2]
```diff
@@ -7,8 +7,33 @@
...
+  fast_check_only: true
+  commands:
+  - apt-get install curl libsodium23 && pytest -v -s tensorizer_loader # Tensorizer
+  - pytest -v -s metrics # Metrics
+  - "pip install \
+      opentelemetry-sdk \
+      opentelemetry-api \
+      opentelemetry-exporter-otlp \
+      opentelemetry-semantic-conventions-ai" # Tracings
+  - pytest -v -s tracing
```
Comment (DarkLight1337) [inline .buildkite/test-pipeline.yaml:32]:
```suggestion
- label: Tensorizer, Metrics, Tracing test
  fast_check: true
  fast_check_only: true
  commands:
  - apt-get install curl libsodium23 && pytest -v -s tensorizer_loader # Tensorizer
  - pytest -v -s metrics # Metrics
  - "pip install \
      opentelemetry-sdk \
      opentelemetry-api \
      opentelemetry-exporter-otlp \
      opentelemetry-semantic-conventions-ai" # Tracing
  - pytest -v -s tracing
```

Comment (khluu) [reply .buildkite/test-pipeline.yaml]: fixed

**Comment by simon-mo** [comment]
* What is fast_check_only?
* Can we add docs build as well.

**Review by simon-mo — APPROVED** [review]

**Comment by Jeffwan** [comment]
@khluu I am new to buildkite tool. I can not find `fast_check` in official docs. Is there something I am missing? https://buildkite.com/docs/pipelines/command-step#command-step-attributes

**Pull request merged**